content/blog/the-world-of-web-rag/index.md

```diff
@@ -0,0 +1,359 @@
+answer the user's question.
+"""
+pipe = Pipeline()
+
+pipe.add_component("search", SerperDevWebSearch(api_key=serperdev_api_key))
+pipe.add_component("prompt_builder", PromptBuilder(template=template))
+pipe.add_component("llm", GPTGenerator(api_key=api_key))
+pipe.connect("search.documents", "prompt_builder.documents")
+pipe.connect("prompt_builder", "llm")
+
+question = "What's the official language of the Republic of Rose Island?"
+pipe.run({
+    "search": {"query": question},
+    "prompt_builder": {"question": question}
+})
+# returns {
+#     'llm': {
+#         'replies': [
+#             "The official language of the Republic of Rose Island is Esperanto. This artificial language was chosen by the residents of Rose Island as their national language when they declared independence in 1968. However, it's important to note that despite having their own language, government, currency, and postal service, Rose Island was never officially recognized as an independent nation by any country."
+#         ],
+#         'metadata': [...]
+#     }
+# }
+```
+
+![Minimal Web RAG Pipeline](minimal-web-rag-pipeline.png)
+
+This solution is already quite effective for simple questions because Google does most of the heavy lifting of reading the content of the top results, extracting the relevant snippets, and packaging them up in a way that is really easy to access and understand by the model.
+
+However, there are situations in which this approach is not sufficient. For example, for highly technical or nuanced questions, the answer box does not provide enough context for the LLM to elaborate and grasp the entire scope of the discussion. In these situations, we may need to turn to the second output of `SerperDevWebSearch`: the links.
+
+# Fetching URLs
+
+Haystack offers components to read the content of a URL: it's `LinkContentFetcher`. Let's see this component in action.
+
+```python
+from haystack.preview.components.fetchers.link_content import LinkContentFetcher
+
+fetcher = LinkContentFetcher()
+fetcher.run(urls=["https://en.wikipedia.org/wiki/Republic_of_Rose_Island"])
+# returns {
+#     "streams": [
+#         ByteStream(data=b"<DOCTYPE html>\n<...")
+#     ]
+# }
+```
+
+First, let's notice that `LinkContentFetcher` outputs a list of `ByteStream` objects. `ByteStream` is a Haystack abstraction that makes handling binary streams and files equally easy. When a component produces `ByteStream` as output, you can directly pass these objects to a Converter component that can extract its textual content without saving such binary content to a file.
+
+These features come in handy to connect `LinkContentFetcher` to a component we've already met before: `HTMLToDocument`.
+
+# Processing the page
+
+In a [previous post](#), we've seen how Haystack can convert web pages into clean Documents ready to be stored in a Document Store. We will reuse many of the components we have discussed there, so if you missed it, make sure to check it out.
```

Comment:
This was linking to a post that we haven't published here yet. LMK what to point it to.

Reply:
Again, same as previous comment, we will probably publish this first so we can remove this line?